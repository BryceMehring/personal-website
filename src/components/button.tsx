import { Link } from './link';

export const Button = (props: ButtonProps): JSX.Element => {
  return (
    <Link className="btn btn-primary m-3" href={props.link}>
      {props.text}
    </Link>
  );
};
