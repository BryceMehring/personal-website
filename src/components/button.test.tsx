import { Button } from './button';
import renderer from 'react-test-renderer';
import { describe, it, expect } from 'vitest';

describe('button', () => {
  it('matches snapshot', () => {
    const wrapper = renderer.create(<Button text="Hello" link="test" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
