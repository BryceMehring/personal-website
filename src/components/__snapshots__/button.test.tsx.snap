// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button > matches snapshot 1`] = `
<a
  className="btn btn-primary m-3"
  href="test"
>
  Hello
</a>
`;
